Add Integrations page for Go (#11360)
getsentry/sentry-docs: docs/platforms/go/common/integrations/index.mdx

---
title: Integrations
description: "Learn about the automatic integrations Sentry provides and how to configure them."
sidebar_order: 500
---

## Default Integrations

These integrations are enabled by default and integrate into the standard library or the Go runtime itself. They are documented so you can be aware of what they do and disable them if they cause issues.

To disable default integrations, you can provide an empty list of integrations when calling `sentry.Init()`.

```go
sentry.Init(sentry.ClientOptions{
    Dsn:               "___PUBLIC_DSN___",
	Integrations: func(i []Integration) []Integration {
		return []Integration{}
	    },
    }
)

> ```

Reviewer: Indentation is off.

Author: Same here:

![image](https://github.com/user-attachments/assets/b3edb525-1383-43ef-9767-8bb53c7f9d7b)

### ModulesIntegration

> This integration logs all the Go modules used in your project, including versions, as part of event details.

Reviewer: ```suggestion
This integration records all Go modules used in your application, including the version, as part of event details.
```

### EnvironmentIntegration

> This integration fills the event data with Go runtime and OS-level information, such as CPU count, Go architecture, and runtime version.

Reviewer: ```suggestion
This integration fills the event data with Go runtime and OS-level specific information, such as CPU core/thread count, Go architecture, and runtime version.
```

Reviewer: Go architecture?

### IgnoreErrorsIntegration

This integration allows you to ignore certain error patterns using regular expressions. It ensures that errors matching the specified patterns are not sent to Sentry.

### IgnoreTransactionsIntegration

This integration lets you ignore specific transactions that match the provided patterns. This is useful when certain transaction names should not be captured.

### ContextifyFramesIntegration

This integration captures context around lines of code that caused an error. It provides a snapshot of source code, including lines before and after the error, for better debugging.

### GlobalTagsIntegration

> This integration adds global tags to all events. These can be defined as environment variables using the SENTRY_TAGS_ prefix or within the client options.

Reviewer: ```suggestion
This integration adds global tags to all events. These can be defined as environment variables using the `SENTRY_TAGS_` prefix or within the client options.
```

### Disabling certain Integrations

You can customize the list of integrations without disabling all the default ones using the Integrations option. In the example below, all integrations are enabled except the ContextifyFrames Integration:

```go
sentry.Init(sentry.ClientOptions{
	Integrations: func(integrations []sentry.Integration) []sentry.Integration {
		var filteredIntegrations []sentry.Integration
		for _, integration := range integrations {
			if integration.Name() == "ContextifyFrames" {
				continue
			}
			filteredIntegrations = append(filteredIntegrations, integration)
		}
		return filteredIntegrations
	},
})

> ```

Reviewer: Indentation is off.

Author: Sorry, not sure how this is off? Running it locally it looks like this:

![image](https://github.com/user-attachments/assets/b50ff20e-a3d1-41c0-97a5-d99b57ad144c)
